c2cgeoportal/__init__.py

```diff
@@ -426,30 +433,59 @@ def default_user_validator(request, username, password):
     return username if user and user.validate_password(password) else None
 
 
-def ogcproxy_route_predicate(info, request):
+class OgcproxyRoutePredicate:
     """ Serve as a custom route predicate function for ogcproxy.
     We do not want the OGC proxy to be used to reach the app's
     mapserv script. We just return False if the url includes
     "mapserv". It is rather drastic, but works for us. """
-    url = request.params.get("url")
-    if url is None:
-        return False
-    if url.find("mapserv") > 0:
-        return False
-    return True
 
+    def __init__(self, val, config):
+        self.private_networks = [
+            Network("127.0.0.0/8"),
+            Network("10.0.0.0/8"),
+            Network("172.16.0.0/12"),
+            Network("192.168.0.0/16"),
+        ]
+
+    def __call__(self, context, request):
+        url = request.params.get("url")
+        if url is None:
+            return False
+
+        parts = urlsplit(url)
+        try:
+            ip = IP(gethostbyname(parts.netloc))
+        except gaierror as e:
+            log.info("Unable to get host name for %s: %s" % (url, e))
+            return False
+        for net in self.private_networks:
+            if ip in net:
+                return False
+        return True
+
+    def phash(self):
```

Comment:
See: http://docs.pylonsproject.org/projects/pyramid/en/1.6-branch/narr/hooks.html#view-and-route-predicates